### ISSUE TITLE
Electron v11 on MacOS: error: no template named 'remove_cv_t' in namespace 'std'; did you mean 'remove_cv'?

### ISSUE THREAD

**Issue by pattrickrice** [microsoft/node-pty#441]
## Environment details

- OS: MacOS
- OS version: 10.15.7
- node-pty version: 0.9.0

## Issue description

```bash
➜  ~ vue --version
@vue/cli 4.5.9
➜  ~ vue create hello-world



Vue CLI v4.5.9
? Please pick a preset: Default ([Vue 2] babel, eslint)


Vue CLI v4.5.9
✨  Creating project in /Users/price/hello-world.
🗃  Initializing git repository...
⚙️  Installing CLI plugins. This might take a while...

yarn install v1.22.4
info No lockfile found.




success Saved lockfile.
✨  Done in 26.14s.
🚀  Invoking generators...
📦  Installing additional dependencies...
[-/5] ⠠ waiting...
yarn install v1.22.4
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
[4/4] 🔨  Building fresh packages...

success Saved lockfile.
✨  Done in 5.27s.
⚓  Running completion hooks...

📄  Generating README.md...

🎉  Successfully created project hello-world.
👉  Get started with the following commands:

 $ cd hello-world
 $ yarn serve

➜  ~ cd hello-world
➜  hello-world git:(master) vue add electron-builder



📦  Installing vue-cli-plugin-electron-builder...

yarn add v1.22.4
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
[4/4] 🔨  Building fresh packages...

success Saved lockfile.
success Saved 77 new dependencies.
info Direct dependencies
└─ vue-cli-plugin-electron-builder@2.0.0-rc.5
info All dependencies
├─ @develar/schema-utils@2.6.5
├─ @npmcli/move-file@1.0.1
├─ @sindresorhus/is@0.14.0
├─ @szmarczak/http-timer@1.1.2
├─ @types/debug@4.1.5
├─ @types/yargs-parser@15.0.0
├─ @types/yargs@15.0.10
├─ ansi-align@3.0.0
├─ app-builder-bin@3.5.10
├─ at-least-node@1.0.0
├─ boxen@4.2.0
├─ cacheable-request@6.1.0
├─ chromium-pickle-js@0.2.0
├─ ci-info@2.0.0
├─ cli-boxes@2.2.1
├─ clone-response@1.0.2
├─ configstore@5.0.1
├─ crypto-random-string@2.0.0
├─ decompress-response@3.3.0
├─ deep-extend@0.6.0
├─ defer-to-connect@1.1.3
├─ dmg-builder@22.9.1
├─ duplexer3@0.1.4
├─ electron-builder@22.9.1
├─ electron-publish@22.9.1
├─ escape-goat@2.1.1
├─ filelist@1.0.1
├─ friendly-errors-webpack-plugin@1.7.0
├─ global-dirs@2.0.1
├─ got@9.6.0
├─ has-yarn@2.1.0
├─ hosted-git-info@3.0.7
├─ http-cache-semantics@4.1.0
├─ immediate@3.0.6
├─ import-lazy@2.1.0
├─ ini@1.3.5
├─ is-installed-globally@0.3.2
├─ is-npm@4.0.0
├─ is-yarn-global@0.3.0
├─ isbinaryfile@4.0.6
├─ jake@10.8.2
├─ json-buffer@3.0.0
├─ jszip@3.5.0
├─ keyv@3.1.0
├─ latest-version@5.1.0
├─ lie@3.3.0
├─ lodash.merge@4.6.2
├─ lowercase-keys@1.0.1
├─ minizlib@2.1.2
├─ p-cancelable@1.1.0
├─ package-json@6.5.0
├─ pumpify@2.0.1
├─ pupa@2.1.1
├─ registry-auth-token@4.2.1
├─ registry-url@5.1.0
├─ responselike@1.0.2
├─ semver-diff@3.1.1
├─ set-immediate-shim@1.0.1
├─ shebang-loader@0.0.1
├─ split2@3.2.2
├─ stat-mode@1.0.0
├─ tar@6.0.5
├─ term-size@2.2.1
├─ through2-filter@3.0.0
├─ through2-map@3.0.0
├─ to-readable-stream@1.0.0
├─ truncate-utf8-bytes@1.0.2
├─ typedarray-to-buffer@3.1.5
├─ unique-string@2.0.0
├─ unzip-crx@0.2.0
├─ update-notifier@4.1.3
├─ url-parse-lax@3.0.0
├─ utf8-byte-length@1.0.4
├─ vue-cli-plugin-electron-builder@2.0.0-rc.5
├─ widest-line@3.1.0
├─ write-file-atomic@3.0.3
└─ yaku@0.16.7
✨  Done in 6.67s.
✔  Successfully installed plugin: vue-cli-plugin-electron-builder

? Choose Electron Version ^9.0.0

🚀  Invoking generator for vue-cli-plugin-electron-builder...
📦  Installing additional dependencies...

yarn install v1.22.4
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
[4/4] 🔨  Building fresh packages...
success Saved lockfile.
$ electron-builder install-app-deps
(node:68360) ExperimentalWarning: The fs.promises API is experimental
  • electron-builder  version=22.9.1
✨  Done in 13.89s.
⚓  Running completion hooks...

✔  Successfully invoked generator for plugin: vue-cli-plugin-electron-builder
➜  hello-world git:(master) ✗ yarn add node-pty
yarn add v1.22.4
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
[4/4] 🔨  Building fresh packages...
success Saved lockfile.
success Saved 1 new dependency.
info Direct dependencies
└─ node-pty@0.9.0
info All dependencies
└─ node-pty@0.9.0
$ electron-builder install-app-deps
(node:68500) ExperimentalWarning: The fs.promises API is experimental
  • electron-builder  version=22.9.1
  • rebuilding native dependencies  dependencies=node-pty@0.9.0 platform=darwin arch=x64
  • rebuilding native dependency  name=node-pty version=0.9.0
✨  Done in 11.91s.
➜  hello-world git:(master) ✗ yarn add electron
yarn add v1.22.4
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
[4/4] 🔨  Building fresh packages...
success Saved lockfile.
warning "electron" is already in "devDependencies". Please remove existing entry first before adding it to "dependencies".
success Saved 1 new dependency.
info Direct dependencies
└─ electron@11.0.1
info All dependencies
└─ electron@11.0.1
$ electron-builder install-app-deps
(node:68662) ExperimentalWarning: The fs.promises API is experimental
  • electron-builder  version=22.9.1
  • rebuilding native dependencies  dependencies=node-pty@0.9.0 platform=darwin arch=x64
  • rebuilding native dependency  name=node-pty version=0.9.0
  ⨯ cannot execute  cause=exit status 1
                    out=yarn run v1.22.4
    $ node scripts/install.js
      CXX(target) Release/obj.target/pty/src/unix/pty.o
    info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

                    errorOut=gyp info it worked if it ends with ok
    gyp info using node-gyp@3.8.0
    gyp info using node@10.14.0 | darwin | x64
    gyp info spawn /usr/bin/python
    gyp info spawn args [ '/Users/price/.nvm/versions/node/v10.14.0/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
    gyp info spawn args   'binding.gyp',
    gyp info spawn args   '-f',
    gyp info spawn args   'make',
    gyp info spawn args   '-I',
    gyp info spawn args   '/Users/price/hello-world/node_modules/node-pty/build/config.gypi',
    gyp info spawn args   '-I',
    gyp info spawn args   '/Users/price/.nvm/versions/node/v10.14.0/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
    gyp info spawn args   '-I',
    gyp info spawn args   '/Users/price/.electron-gyp/11.0.1/include/node/common.gypi',
    gyp info spawn args   '-Dlibrary=shared_library',
    gyp info spawn args   '-Dvisibility=default',
    gyp info spawn args   '-Dnode_root_dir=/Users/price/.electron-gyp/11.0.1',
    gyp info spawn args   '-Dnode_gyp_dir=/Users/price/.nvm/versions/node/v10.14.0/lib/node_modules/npm/node_modules/node-gyp',
    gyp info spawn args   '-Dnode_lib_file=/Users/price/.electron-gyp/11.0.1/<(target_arch)/node.lib',
    gyp info spawn args   '-Dmodule_root_dir=/Users/price/hello-world/node_modules/node-pty',
    gyp info spawn args   '-Dnode_engine=v8',
    gyp info spawn args   '--depth=.',
    gyp info spawn args   '--no-parallel',
    gyp info spawn args   '--generator-output',
    gyp info spawn args   'build',
    gyp info spawn args   '-Goutput_dir=.' ]
    gyp info spawn make
    gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
    In file included from ../src/unix/pty.cc:20:
    In file included from ../../nan/nan.h:56:
    In file included from /Users/price/.electron-gyp/11.0.1/include/node/node.h:67:
    In file included from /Users/price/.electron-gyp/11.0.1/include/node/v8.h:30:
    /Users/price/.electron-gyp/11.0.1/include/node/v8-internal.h:418:38: error: no template named 'remove_cv_t' in namespace 'std'; did you mean 'remove_cv'?
                !std::is_same<Data, std::remove_cv_t<T>>::value>::Perform(data);
                                    ~~~~~^~~~~~~~~~~
                                         remove_cv
    /Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/type_traits:697:50: note: 'remove_cv' declared here
    template <class _Tp> struct _LIBCPP_TEMPLATE_VIS remove_cv
                                                     ^
    1 error generated.
    make: *** [Release/obj.target/pty/src/unix/pty.o] Error 1
    gyp ERR! build error
    gyp ERR! stack Error: `make` failed with exit code: 2
    gyp ERR! stack     at ChildProcess.onExit (/Users/price/.nvm/versions/node/v10.14.0/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:262:23)
    gyp ERR! stack     at ChildProcess.emit (events.js:182:13)
    gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:240:12)
    gyp ERR! System Darwin 19.6.0
    gyp ERR! command "/Users/price/.nvm/versions/node/v10.14.0/bin/node" "/Users/price/.nvm/versions/node/v10.14.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
    gyp ERR! cwd /Users/price/hello-world/node_modules/node-pty
    gyp ERR! node -v v10.14.0
    gyp ERR! node-gyp -v v3.8.0
    gyp ERR! not ok
    error Command failed with exit code 1.

                    command=/Users/price/.nvm/versions/node/v10.14.0/bin/node /usr/local/Cellar/yarn/1.22.4/libexec/bin/yarn.js run install
                    workingDir=/Users/price/hello-world/node_modules/node-pty
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/add for documentation about this command.
```

**Comment by LabhanshAgrawal** [comment]
It works if I replace the remove_cv_t with remove_cv
I'm not familiar with native modules so don't know if it breaks something else.
@Tyriar can you please check it out.

**Comment by qftgtr** [comment]
I think it's because `remove_cv_t` is only supported since C++14 ([ref](https://en.cppreference.com/w/cpp/types/remove_cv)). If I change [binding.gyp#L72](https://github.com/microsoft/node-pty/blob/0.9.0/binding.gyp#L72) to `"-std=c++14"` I can build it successfully.

**Comment by npezza93** [comment]
I think this got resolved in https://github.com/microsoft/node-pty/pull/433. If you pull the latest beta(0.10.0-beta18), it works beautifully on electron 11.

**Comment by starpit** [comment]
Curious: how does this intersect with the need for context-aware native modules? ref: https://github.com/microsoft/node-pty/issues/405

I thought that Electron 11 finally removed support for non-context-aware native modules (which, according to the referenced issue, node-pty is one)?

**Comment by pattrickrice** [comment]
I believe they're removing support for non-context aware modules in the renderer thread, not the main thread. I think in electron v10 if you use node-pty in the renderer you'll see some warnings in the console.

**Comment by starpit** [comment]
I see. Indeed we see those warnings.

**Comment by Tyriar** [comment]
Closing due to https://github.com/microsoft/node-pty/issues/441#issuecomment-735252824

**Comment by MaxWeisen** [comment]
> I think this got resolved in #433. If you pull the latest beta(0.10.0-beta18), it works beautifully on electron 11.

My team is still getting the error mentioned in the title with the following:
- Electron 11
- Node-pty v0.10.0
- Using Electron Rebuild package

Has anyone been able to pinpoint a solution that doesn't require editing the node-pty files after the user has installed?

**Comment by stevecondylios** [comment]
I got the same message (`error: no template named 'remove_cv_t' in namespace 'std'; did you mean 'remove_cv'` when running `yarn install`. For any cpp noobies like me, the key was to run it with the `-std=c++17` argument, i.e. like so: 

```sh
yarn install -std=c++17
```
